packages/shared/src/permissions/permissionsClass.ts

```diff
@@ -0,0 +1,100 @@
+import { MetricInterface } from "back-end/types/metric";
+import { Permission, UserPermissions } from "back-end/types/organization";
+import { READ_ONLY_PERMISSIONS } from "./permissions.utils";
+class PermissionError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "PermissionError";
+  }
+}
+
+export class Permissions {
+  private userPermissions: UserPermissions;
+  private superAdmin: boolean;
+  constructor(permissions: UserPermissions, superAdmin: boolean) {
+    this.userPermissions = permissions;
+    this.superAdmin = superAdmin;
+  }
+
+  public canCreateMetric = (
+    metric: Pick<MetricInterface, "projects">
+  ): boolean => {
+    return this.checkProjectFilterPermission(metric, "createMetrics");
+  };
+
+  public canUpdateMetric = (
+    existing: Pick<MetricInterface, "projects" | "managedBy">,
+    updates: Pick<MetricInterface, "projects">
+  ): boolean => {
+    return this.checkProjectFilterUpdatePermission(
+      existing,
+      updates,
+      "createMetrics"
+    );
+  };
+
+  public canDeleteMetric = (
+    metric: Pick<MetricInterface, "projects" | "managedBy">
+  ): boolean => {
+    return this.checkProjectFilterPermission(metric, "createMetrics");
+  };
+
+  public throwPermissionError(): void {
+    throw new PermissionError(
+      "You do not have permission to perform this action"
+    );
+  }
+
+  private checkProjectFilterPermission(
+    obj: { projects?: string[] },
+    permission: Permission
+  ): boolean {
+    const projects = obj.projects?.length ? obj.projects : [""];
+
```

Comment:
We'll need to add logic here for `READ_ONLY_TYPE` permissions e.g. `readData`, `addComments`, etc, but will wait to do so until it's relevant.